maven-xml/src/main/java/org/apache/maven/xml/filter/DependencyKey.java

```diff
@@ -0,0 +1,88 @@
+package org.apache.maven.xml.filter;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Objects;
+
+/**
+ * 
+ * @author Robert Scholte
+ * @since 3.7.0
+ */
+public class DependencyKey
+{
+    private final String groupId;
+    
+    private final String artifactId;
+
+    public DependencyKey( String groupId, String artifactId )
+    {
+        this.groupId = groupId;
+        this.artifactId = artifactId;
```

Comment:
Compute the hash there since it is a hadhmap key ;)

Reply:
good catch

Reply:
Done